spec/controllers/course/admin/admin_controller_spec.rb

```diff
@@ -111,9 +111,9 @@
       context 'when the user is a Course Manager' do
         let(:user) { create(:course_manager, course: course).user }
 
-        it 'destroys the course' do
+        it 'soft deletes the course' do
```

Comment:
Why do we need course soft-deletion in this PR?

Reply:
Because we apply the soft-delete for CourseUser, which depend on Course.
We also allow to delete the Course, but if the Course is deleted, the CourseUser will need hard-delete to avoid "FK" violation.
So we need soft-delete for Course also.

Reply:
We can hard-delete CourseUser if associated Course is deleted.  
However, if we want to implement soft-delete for Course, we should also ensure that the soft-deletion is performant + there must be a way to restore the course. We can do this in a separate PR.

Reply:
The soft-delete for Course PR might need come first this PR, otherwise, a few of tests (which delete the Course) will be failed.

Reply:
> We can hard-delete CourseUser if associated Course is deleted.